src/client/dfs/duns.c

```diff
@@ -442,16 +445,53 @@ duns_resolve_path(const char *path, struct duns_attr_t *attr)
 	/** no match for direct format, do the UNS fs check */
 
 	rc = statfs(path, &fs);
+
+#ifdef LUSTRE_INCLUDE
+	/* since statfs follows symlinks, may need to use directory to
+	 * determine FS type, if on Lustre and path is a foreign symlink
+	 */
+	if (rc == -1 || fs.f_type != LL_SUPER_MAGIC) {
+		int		errno_save = errno;
+		struct statfs	fs2;
+		char		*dirp;
+		int		rc2;
+
+		D_STRNDUP(dir, path, PATH_MAX);
+		if (dir == NULL) {
+			D_ERROR("Failed to copy path\n");
+			return ENOMEM;
+		}
```

Comment:
No need to log here, and therefore no reason for the brackets.
```suggestion
		if (dir == NULL)
			return ENOMEM;
```

Reply:
ok, done